R/excitation.R

```diff
@@ -0,0 +1,147 @@
+#' Excitation
+#'
+#' @description Excitation lets you add visual citations as tooltips to your
+#'   \pkg{xaringan} slides. Simply include a `.bib` file in your R Markdown
+#'   YAML header, as usual, and cite away!
+#'
+#' @section Usage: To use excitation for visual citations, add the following
+#'   code chunk to your slides' R Markdown file.
+#'
+#'   ````markdown
+#'   ```{r xaringan-excitation, echo=FALSE}
+#'   xaringanExtra::use_excitation()
+#'   ```
+#'   ````
+#'
+#' @examples
+#' use_excitation()
+#'
+#' @name excitation
+NULL
+
+#' @describeIn excitation Adds visual citations to your xaringan slides.
+#' @export
+use_excitation <- function() {
+  assign('.excitation_keys', vector('character'), envir = globalenv())
+  htmltools::tagList(
+    html_dependency_popper(),
+    html_dependency_tippy(),
+    html_dependency_excitation()
+  )
+}
+
+#' @describeIn excitation Adds visual citations to your xaringan slides.
+#' @export
+excite <- function(key) {
+  bibs <- rmarkdown::metadata$bibliography
+  self_contained <- FALSE
+
+  if (is.null(bibs)) return(key)
+
+  if (length(bibs) > 1) {
+    tmp_bib <- tempfile(fileext = '.bib')
+    for (i in seq_len(bibs)) {
+      write(readLines(bibs[[i]]), tmp_bib, append = TRUE)
+    }
+    bibs <- tmp_bib
+  }
+
+  .excitation_keys <- get('.excitation_keys', envir = globalenv())
+
+  script <- ""
+
+  if (!key %in% .excitation_keys) {
+    tooltip <- htmltools::includeHTML(
+      namedropR::drop_name(
```

Comment:
It's unfortunate that `namedropR::drop_name()` prints messages, but we should muffle them and only report errors.

```
Years coerced to string format.
Bibliography file successfully read.
```